README.md

```diff
@@ -1 +1,144 @@
-# rfq-marketplace
\ No newline at end of file
+# rfq-marketplace
+
+rfq-marketplace is a contract that provides a token exchange platform.
+
+## Prerequisites
+
+- Node.js
+- NPM
+
+## Deploy the contract
+
+The first step, clone the project and install the dependency.
+```
+git clone https://github.com/second-state/rfq-marketplace.git
+cd rfq-marketplace
+npm install
+```
+You can edit the `hardhat.config.js` to choose which network and account to deploy the contract.
+In the example, we use CyberMiles chain to demo.
+
+Next step, run the following command to run the deploy script.
+```
+npx hardhat run scripts/deploy.js --network cybermiles
+```
+
+Then you can see the contract address.
+```
+Deployed to 0x21b35E5D3689fF5C6cc1FDF09c1B13438007FaF1
+```
+
+## Deploy web service on flows.network
+
+To interact with rfq-marketplace contract, we will use [flows.network](https://flows.network/), a serverless platform that makes deploying your own app quick and easy in just three steps.
+
+### Prerequisite
+
+You will need a wallet private key. If you do not already have one, use [Metamask](https://metamask.io/) to create it.
+
+#### Fork this repo and write your own code
+
+Fork [this repo](https://github.com/second-state/rfq-marketplace.git). 
+
+#### Deploy the code on flows.network
+
+1. Sign up for an account for deploying flows on [flows.network](https://flows.network/). It's free.
+2. Click on the "Create a Flow" button to start deploying the web service.
+3. Authenticate the [flows.network](https://flows.network/) to access the `rfq-marketplace` repo you just forked. 
+
+4. Click on the Advanced text and you will see more settings including branch and environment variables. The function code is stored in `flows` folder, you need to change Directory to `/flows`.
+In this example, we have three variables to fill in, `CONTRACT_ADDRESS` is rfq-marketplace contract [address]( #deploy-the-contract).
+The default network is cybermiles. If you want to change the network, you can set `RPC_NODE_URL` and `CHAIN_ID` variable.
+
+<img width="899" alt="image" src="https://i.imgur.com/xSAxwLF.png">
+
+5. Click the Deploy button to deploy your function.
+
+### Configure SaaS integrations
+
+After that, the flows.network will direct you to configure the SaaS integration required by your flow. Here we can see: there is no SaaS needs to be connected since it's a lambda service. Just click the Check button to see your flow details.
+
+<img width="964" alt="image" src="https://user-images.githubusercontent.com/45785633/226959151-0e8a159a-02b3-4130-b7b5-8831b65c8d75.png">
+
+## Try this demo
+### Create an exchange request
+
+You can use the `submit-request` function to create a new exchange request. You need to provide three query parameters.<br>
+`tokenA`: The token address you want to exchange out. <br>
```

Comment:
Let's use `token_in` and `token_out` as parameter names for clarifty.

Reply:
I have changed it. Please check, thank you.